File: benchmarks/ruby/string_ops.rb
ITERATIONS = 1000

start_time = Time.now

result = ""
(0...ITERATIONS).each do |i|
    result += "Hello#{i} "
end

end_time = Time.now

puts "Iterations: #{ITERATIONS}"
puts "Result length: #{result.length}"
puts "Time: #{((end_time - start_time) * 1000).round(2)}ms"

File: benchmarks/javascript/string_ops.js
const ITERATIONS = 1000;

const startTime = Date.now();

let result = "";
for (let i = 0; i < ITERATIONS; i++) {
    result += `Hello${i} `;
}

const endTime = Date.now();

console.log(`Iterations: ${ITERATIONS}`);
console.log(`Result length: ${result.length}`);
console.log(`Time: ${endTime - startTime}ms`);